.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,30 @@
+name: Pre-commit actions for Cocuhbase Plugin
+
+on: [pull_request]
+
+jobs:
+  pytest:
+    name: Test ${{ matrix.package }} on ${{ matrix.os }} using pytest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [2.7]
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
I think you could test on ubuntu or mac based on your local developer environments, unless you all use windows too or you want to support external contributors from plugin authors that have a local windows setup for plugin development?
The SDK has to support all 3 since we support plugin developers on all platforms. This might or might not be the appropriate choice for couchbase. Supporting additional environments also comes with an additional development cost.

Reply:
> Could you update the PR description to add information about what this PR does?
> 
> Question: I didn't see the checks get kicked off for this PR, my understanding was that if checks are added in a PR, they run for that PR too? Just checking if it works differently.
Please correct me if wrong. My understanding is that this check will apply only this 'pre-commit.yml'  file is present on the destination branch( In this case Delphix master branch).
I was unable to test locally because more than one fork for a user is restricted for one branch.

Reply:
> I think you could test on ubuntu or mac based on your local developer environments, unless you all use windows too or you want to support external contributors from plugin authors that have a local windows setup for plugin development?
> The SDK has to support all 3 since we support plugin developers on all platforms. This might or might not be the appropriate choice for couchbase. Supporting additional environments also comes with an additional development cost.

Yes, that's a good point. It will add additional development costs,  so we are removing the dependency from the windows platform.